app/models/user.rb

```diff
@@ -9,4 +9,40 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :requested_friendships, foreign_key: 'requester_id', class_name: 'Friendship'
+  has_many :received_friendships, foreign_key: 'receiver_id', class_name: 'Friendship'
+
+  #  has_many :friendships
+  #  has_many :inverse_friendships, :class_name => "Friendship", :foreign_key => "friend_id"
+
+  def friends
+    requested_array = requested_friendships.map { |fr| fr.receiver if fr.status }
+    received_array = received_friendships.map { |fr| fr.requester if fr.status }
+    (requested_array + received_array).compact
+  end
+
+  def friendships
+    requested = requested_friendships
+    received = received_friendships
+    (requested + received)
+  end
```

Comment:
Same with this one, you'll only need to search for either the requested or received friendship, no need to the two queries.